### PR TITLE
vectorised first set bit in bitmap containers

### PR DESCRIPTION
**Pull request by richardstartin**
Uses multi release jar feature to allow vectorised scan for the first set bit in a `BitmapContainer`, improves throughput of `BitmapContainer.first` 2-3x. 

Does allocate a 2KB static final `long[]` however, interested in opinions on whether this is deemed an acceptable cost.

### CODE REVIEW

**Comment by lemire** [comment]
Do you need the array to be that big? Could it be smaller?

**Comment by richardstartin** [comment]
Let me investigate, obviously this is not cache efficient.

**Comment by richardstartin** [comment]
Reducing the size of the array to 64 longs (0.5kB) seems to improve the performance slightly. At 0.5kB is it an acceptable trade off?

**Comment by lemire** [comment]
Yes. I think it is. If you are ok with the current PR, I suggest we merge it.

File: shims/src/java11/java/org/roaringbitmap/ArraysShim.java [hunk 1]
```diff
@@ -20,4 +22,18 @@
...
+  /**
+   * Find the index of the first non empty word
+   *
+   * @param bitmap the bitmap
+   * @return the index of the first non empty word, or -1 if the bitmap is empty
+   */
+  public static int indexOfFirstNonEmptyWord(long[] bitmap) {
+    int firstNonEmpty = -1;
+    for (int i = 0; i < bitmap.length && firstNonEmpty == -1; i += EMPTY.length) {
+      firstNonEmpty = Arrays.mismatch(bitmap, i, i + EMPTY.length, EMPTY, 0, EMPTY.length);
```
[the remark this answers is not in the document]
Comment (richardstartin) [reply shims/src/java11/java/org/roaringbitmap/ArraysShim.java]: I doubt it - `Arrays.mismatch` is an intrinsic method introduced in JDK9 which performs vectorised comparisons, i.e. in parallel. Whether this is actually faster than unrolled scalar comparisons, I don't know, but I would be surprised if the scalar loop weren't unrolled automatically.

**Comment by lemire** [comment]
@richardstartin Please confirm that you want this PR merged. :-)

**Comment by richardstartin** [comment]
@lemire I want to do a bit more testing, I fear this might shine in a benchmark but cause harm at a larger scale by wasting cache space.

**Comment by richardstartin** [comment]
This seems to be one of those things which looks good in benchmarks but not in actual use cases, closing the PR.

**Pull request closed without merging**